wireguard/wireguard.go

```diff
@@ -1498,7 +1498,7 @@ func (w *Wireguard) ensureLinkAddressV4(netlinkClient netlinkshim.Interface) err
 func (w *Wireguard) addRouteRule() {
 	w.routerule.SetRule(routerule.NewRule(ipVersion, w.config.RoutingRulePriority).
 		GoToTable(w.config.RoutingTableIndex).
-		MatchFWMarkWithMask(0, uint32(w.config.FirewallMark)))
+		Not().MatchFWMarkWithMask(uint32(w.config.FirewallMark), uint32(w.config.FirewallMark)))
```

Comment:
Worth a comment just to remind ourselves that there's an issue with matching 0 for the mark, I think.